jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -970,38 +971,49 @@ public boolean checkVirtualHost(final Request baseRequest)
         if (_vhosts != null && _vhosts.length > 0)
         {
             String vhost = normalizeHostname(baseRequest.getServerName());
-
-            boolean match = false;
-            boolean connectorName = false;
-            boolean connectorMatch = false;
-
-            for (String contextVhost:_vhosts)
+            String connectorName = baseRequest.getHttpChannel().getConnector().getName();
+            for (String contextVhost : _vhosts)
             {
-                if (contextVhost == null || contextVhost.length()==0)
+                if (contextVhost == null || contextVhost.length() == 0)
```

Comment:
let's remove the need for this check by preprocessing in the setter (see below)

Reply:
I agree, will fix.